Cargo.toml

```diff
@@ -17,6 +17,7 @@ net = ["futures-lite", "async-net", "async-trait"]
 tls = ["rust_tls"]
 rust_tls = ["net", "rustls", "webpki", "fluvio-async-tls", "pin-project"]
 native2_tls = ["net","pin-project","async-native-tls","native-tls","openssl"]
+openssl_vendored = ["openssl/vendored"]
```

Comment:
`openssl` should be optional feature native2_tls for mac only

Reply:
I don't think you can do target specific features.

Reply:
see: https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html.

```
[target.'cfg(unix)'.dependencies]
openssl = "1.0.1"
```

Reply:
When I run `make build-all` on my linux machine I get a bunch of openssl errors. Or are you suggesting we move the native_tls as a mac only module in this PR?

Reply:
> Or are you suggesting we move the native_tls as a mac only module in this PR?

I did that.

Reply:
Hmm.  actually, other way around,  openssl/vendor should be only used on Linux.   native_tls works on iOS without openssl

Reply:
should follow https://crates.io/crates/native-tls

Reply:
We need openssl for the native2_tls feature becuase openssl is used in `src/native_tls.rs`, regardless of platform

Reply:
I dont think we can expect openssl to be installed on iOS, therefore we must vendor. Alternative is to use the `x509-parser` crate instead for certificate manupilation logic. Then we can remove `openssl` as a dependency of `native2_tls`

Reply:
After looking into this a bit more, I think we should definitely remove `openssl` as a dependency because android support will be a similar issue. I was hoping it'd be something I could easily do but I'm willing to admit that I don't know enough about TLS stuff to do this easily and correctly. For now, I think we might be able to simple make `openssl` be vendored for iOS targets by default as a work around to allow for building out support for swift. What do you guy think?

Reply:
Yes, for short-term. let's remove explicit `openssl` dependencies and let downstream crates to vendor it.
We should add feature flag for openssl dependent X509 handling in the future.